### PR TITLE
Add attributes to video stream metacards

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Add attributes to video stream metacards
- Add start, stop, duration, and bitrate
- Add recording-id and segment-ids 
Simplified synchronization in PacketBuffer
#### Who is reviewing it? 

@jaymcnallie
@jrnorth 
#### How should this be tested?
Create and ingest a UDP stream
Verify video metacards contain new attributes

### CODE REVIEW

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/TimerTaskStreamCreationPlugin.java [hunk 1]
```diff
@@ -38,7 +42,11 @@ private TimerTask createTimerTask(UdpStreamProcessor udpStreamProcessor) {
     return new TimerTask() {
       @Override
       public void run() {
-        udpStreamProcessor.checkForRollover();
+        try {
+          udpStreamProcessor.checkForRollover();
+        } catch (Exception e) {
+          LOGGER.error("Exception running rollover check", e);
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/TimerTaskStreamCreationPlugin.java:48]: Seems like this should be `debug`.

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java [hunk 2]
```diff
@@ -170,6 +175,25 @@ public MetacardImpl doAction(MetacardImpl metacard, File tempFile)
...
+    Date start = new Date(startTime);
+    Date end = new Date(endTime);
+
+    metacard.setAttribute(DateTime.START, start);
+    metacard.setAttribute(DateTime.END, end);
+    metacard.setAttribute(Media.DURATION, TimeUnit.MILLISECONDS.toSeconds(endTime - startTime));
+    metacard.setAttribute(
+        Media.BITS_PER_SECOND,
+        (segmentFile.length() * 8.0)
+            / TimeUnit.MILLISECONDS.toSeconds(end.getTime() - start.getTime()));
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java:194]: ✏️ `endTime` - `startTime`

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverActionTest.java [hunk 3]
```diff
@@ -114,6 +117,11 @@ public void setup() throws SourceUnavailableException, IngestException {
     when(udpStreamProcessor.getStreamId()).thenReturn(streamId);
     when(udpStreamProcessor.getSubject()).thenReturn(new SimpleSubject());
 
+    packetBuffer = mock(PacketBuffer.class);
+    when(packetBuffer.getLastSegmentStart()).thenReturn(-1l);
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverActionTest.java:121]: We should have a test for the happy path where both times are positive.

Comment (clockard) [reply catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverActionTest.java:126]: Added test

**Comment by clockard** [comment]
build now

**Review by jrnorth — APPROVED** [review]

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java [hunk 4]
```diff
@@ -213,13 +226,30 @@ private String generateFilename() {
...
+    if (parent != null) {
+      Query query = new QueryImpl(filterBuilder.attribute(Core.ID).equalTo().text(parent.getId()));
+      try {
+        parent =
+            catalogFramework.query(new QueryRequestImpl(query)).getResults().stream()
+                .map(Result::getMetacard)
+                .findFirst()
+                .orElse(parent);
+      } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+        LOGGER.warn("Unable to get fresh parent metacard for {}", parent.getId(), e);
```
Comment (jrnorth) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/rollover/CatalogRolloverAction.java:247]: Probably should be `debug`

**Review by jrnorth — APPROVED** [review]

**Comment by clockard** [comment]
Jenkins is having a problem with the fabric docker plugin when creating the docker image. The same error is occurring on master. This isn't a problem with local builds. 
Merging.

**Pull request merged**